crates/nargo/src/cli/verify_cmd.rs

```diff
@@ -32,9 +33,22 @@ fn process_abi_with_verifier_input(
     abi: Abi,
     pi_map: BTreeMap<String, InputValue>,
 ) -> Result<Vec<FieldElement>, CliError> {
-    let mut public_inputs = Vec::with_capacity(pi_map.len());
+    // Filter out any private inputs from the ABI.
+    let public_abi = abi.public_abi();
 
-    for (param_name, param_type) in abi.parameters.into_iter() {
+    // Check that enough public params were supplied.
+    let num_pub_params = public_abi.num_parameters();
+    if num_pub_params != pi_map.len() {
+        return Err(CliError::Generic(format!(
+            "Expected {} number of values, but got {} number of values",
```

Comment:
Since you are refining cliError, you could add an AbiMismatch error for this case
Move the check after the parameter names (cf. comment in prove_cmd)